components/scream/docs/source-tree.md

```diff
@@ -0,0 +1,63 @@
+# SCREAM's Source Tree
+
+All SCREAM-specific code can be found in `components/scream` within the
+[SCREAM repo](https://github.com/E3SM-Project/scream). Here's how things are
+organized:
+
++ `bin`: Miscellaneous tools for helping with the build process. Currently, this
+  directory only contains a template for generating the `scream_mpicxx` compiler
+  script.
++ `cime_config`: Tools and XML files for integrating SCREAM with E3SM via the
+  CIME framework.
++ `cmake`: CMake functions and macros used by the configuration/build system.
++ `data`: Data files used by our tests.
++ `docs`: Documentation for the SCREAM project, including design documents,
+  instructions for building and testing SCREAM, and this document.
++ `extern`: Source for certain lightweight third-party libraries, embedded
+  directly into the repo (and not imported as submodules).
++ `scripts`: Miscellaneous scripts that implement workflows for running tests
+  and analyzing performance.
++ `src`: All C++ source code (and any bridges to Fortran) for SCREAM are stored
+  here. We describe the contents of this directory in greater detail below.
++ `tests`: Implements standalone, end-to-end tests for various SCREAM
+  components (RRTMG, HOMME, P3, SHOC, etc).
+
+In addition, you'll notice the following files in `components/scream`:
+
++ `CMakeLists.txt`: The CMake file that defines SCREAM's configuration/build
+  system.
++ `CTestConfig.cmake`: This CTest file contains parameters that determine how
+  our test results are reported to the [E3SM CDash Site](http://my.cdash.org/submit.php?project=E3SM).
++ `README.md`: SCREAM's top-level README file, which describes the project and
+  its purpose.
+
+## The `src` Directory
+
+Herein lіes the source code for SCREAM. Broadly, here's where things are:
+
++ `control`: Contains the atmosphere driver and basic tests for it.
```

Comment:
@bartgol and @AaronDonahue - would 'driver' be a better name for the directory containing the AD? I don't think we need to replicate CAM wording if it doesn't make sense. This isn't something we should (?) do in this PR...

Reply:
I think I picked 'control' cause that would allow me to type `c<TAB>` and get autocompletion, while `d<TAB>` didn't, because of the presence of the dynamics folder as well. And having to type `dy<TAB>` to get in the dynamics folder was annoying, since the 'y' key is one of the most unconvenient to type.

That's incidentally the same reason why I picked 'interface' rather than 'coupling' or 'coupler' for the code related to the component coupler. It would clash with 'control'.

#lazy

Reply:
not sure how to respond to that. "Laziness is a virtue"? Let's leave things as-is for now. I think having similar objects with different names will make onboarding harder for domain scientists looking at the code for the first time, so I may still want to change things later.